library/sets/tests_custom.lgt

```diff
@@ -0,0 +1,101 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+%  This file is part of Logtalk <https://logtalk.org/>
+%  Copyright 1998-2022 Paulo Moura <pmoura@logtalk.org>
+%  SPDX-License-Identifier: Apache-2.0
+%
+%  Licensed under the Apache License, Version 2.0 (the "License");
+%  you may not use this file except in compliance with the License.
+%  You may obtain a copy of the License at
+%
+%      http://www.apache.org/licenses/LICENSE-2.0
+%
+%  Unless required by applicable law or agreed to in writing, software
+%  distributed under the License is distributed on an "AS IS" BASIS,
+%  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%  See the License for the specific language governing permissions and
+%  limitations under the License.
+%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+:- object(reverse_int, implements(comparingp)).
+
+    :- public(n/1).
+    :- public(valid/1).
+    valid(X) :-
+        X::n(N),
+        integer(N).
+
+    X > Y :-
+        X::n(N0),
+        Y::n(N1),
+        { N0 < N1 }.
+
+    X < Y :-
+        X::n(N0),
+        Y::n(N1),
+        { N0 > N1 }.
+
+    X =:= Y :-
+        X::n(N0),
+        Y::n(N1),
+        { N0 =:= N1 }.
+
+:- end_object.
+
+:- object(r1, extends(reverse_int)).
+    n(1).
+:- end_object.
+
+:- object(r2, extends(reverse_int)).
+    n(1).
+:- end_object.
+
+:- object(r3, extends(reverse_int)).
+    n(50).
+:- end_object.
+
+:- object(r4, extends(reverse_int)).
+    n(100).
+:- end_object.
+
+:- object(tests_custom,
+	extends(lgtunit)).
+
+    cover(set(reverse_int)).
+
+    % as_set/2 tests
+
+    test(as_set_2_01, deterministic(N == 3)) :-
+        set(reverse_int)::as_set([r1, r2, r3, r4], Set),
+        length(Set, N).
```

Comment:
We cannot assume that the `length/2` predicate is available as a built-in predicate. We can either write `list::length(…)` or use a `uses/2` directive. Given the other calls, I suggest the former.

Reply:
P.S. You can set the `portability` flag to `warning` to detect these issues.

Reply:
TIL about `length/2` not being a built-in